lib/srv/db/mysql/mysql_activate_user.sql

```diff
@@ -0,0 +1,60 @@
+CREATE PROCEDURE teleport_activate_user(IN username VARCHAR(32), IN details JSON)
+proc_label:BEGIN
+    DECLARE is_auto_user INT DEFAULT 0;
+    DECLARE is_active INT DEFAULT 0;
+    DECLARE role_index INT DEFAULT 0;
+    DECLARE role VARCHAR(32) DEFAULT '';
+    SET @roles = details->"$.roles";
+
+    -- If the user already exists and was provisioned by Teleport, reactivate
+    -- it, otherwise provision a new one.
+    SELECT COUNT(TO_USER) INTO is_auto_user FROM mysql.role_edges WHERE FROM_USER = 'teleport-auto-user' AND TO_USER = username;
+    IF is_auto_user = 1 THEN
+        SELECT COUNT(USER) INTO is_active FROM information_schema.processlist WHERE USER = username;
+
+        -- If the user has active connections, just use it to avoid messing up
+        -- its existing roles.
+        IF is_active = 1 THEN
+            LEAVE proc_label;
+        END IF;
+
+        -- Otherwise reactivate the user, but first strip if of all roles to
+        -- account for scenarios with left-over roles if database agent crashed
+        -- and failed to cleanup upon session termination.
+        CALL teleport_revoke_roles(username);
+
+        -- Ensure the user is unlocked. User is locked at deactivation. 
+        SET @sql := CONCAT_WS(' ', 'ALTER USER', QUOTE(username), 'ACCOUNT UNLOCK');
+        PREPARE stmt FROM @sql;
+        EXECUTE stmt;
+        DEALLOCATE PREPARE stmt;
+    ELSE
+        SET @sql := CONCAT_WS(' ', 'CREATE USER', QUOTE(username), details->>"$.auth_options", 'ATTRIBUTE', QUOTE(details->"$.attributes"));
```

Comment:
I wonder if we can we automatically lock the user after the current user cert expires?

Reply:
we probably can if `event_scheduler` is enabled and we can schedule a procedure call.

Though, I recently learned that the expiration Database Agent gets isn't the same from `tsh login`. `proxyserver.go` needs to call `SignDatabaseCSR` to generate a new cert with database CA. And instead of using the original cert expires, TTL is recalculated based on roles. `proxyserver.go`  does monitor the connection using the original cert expiration.

Reply:
It would be nice, but sounds like the portability might be an issue. I guess we could have it as an optional feature for the future PR.